File: examples/node/neutral_ipc_template/NeutralIpcConfig.js
/**
 * Configuration module for Neutral IPC client.
 * Reads configuration from /etc/neutral-ipc-cfg.json or uses default values.
 * https://github.com/FranBarInstance/neutral-ipc
 */

const fs = require('fs');

class NeutralIpcConfig {
    static HOST = '127.0.0.1';
    static PORT = 4273;
    static TIMEOUT = 10000; // milliseconds
    static BUFFER_SIZE = 8192;
    static CONFIG_FILE = '/etc/neutral-ipc-cfg.json';

    static loadConfig() {
        if (!fs.existsSync(this.CONFIG_FILE)) {
            return {};
        }
        try {
            return JSON.parse(fs.readFileSync(this.CONFIG_FILE, 'utf8'));
        } catch (e) {
            return {};
        }
    }

    static getConfigValue(configDict, key, defaultValue) {
        const value = configDict[key];
        if (value === undefined) {
            return defaultValue;
        }
        if (key === 'host' && typeof value === 'string') {
            return value;
        }
        if (['port', 'timeout', 'buffer_size'].includes(key) && typeof value === 'number') {
            return value;
        }
        return defaultValue;
    }

    static getHost() {
        const config = this.loadConfig();
        return this.getConfigValue(config, 'host', this.HOST);
    }

    static getPort() {
        const config = this.loadConfig();
        return this.getConfigValue(config, 'port', this.PORT);
    }

    static getTimeout() {
        const config = this.loadConfig();
        return this.getConfigValue(config, 'timeout', this.TIMEOUT);
    }

    static getBufferSize() {
        const config = this.loadConfig();
        return this.getConfigValue(config, 'buffer_size', this.BUFFER_SIZE);
    }
}

// Module exports for direct access
module.exports.HOST = NeutralIpcConfig.getHost();
module.exports.PORT = NeutralIpcConfig.getPort();
module.exports.TIMEOUT = NeutralIpcConfig.getTimeout();
module.exports.BUFFER_SIZE = NeutralIpcConfig.getBufferSize();
module.exports.NeutralIpcConfig = NeutralIpcConfig;

File: examples/go/neutral_ipc_template/neutral_ipc_config.go
package neutral_ipc_template

import (
	"encoding/json"
	"os"
)

// Default values
const (
	DefaultHost    = "127.0.0.1"
	DefaultPort    = 4273
	DefaultTimeout = 10
	DefaultBufSize = 8192
)

// ConfigFile is the path to the IPC server configuration file.
const ConfigFile = "/etc/neutral-ipc-cfg.json"

// LoadConfig loads configuration from the JSON file if it exists.
func LoadConfig() map[string]interface{} {
	if _, err := os.Stat(ConfigFile); os.IsNotExist(err) {
		return make(map[string]interface{})
	}

	file, err := os.Open(ConfigFile)
	if err != nil {
		return make(map[string]interface{})
	}
	defer file.Close()

	var config map[string]interface{}
	if err := json.NewDecoder(file).Decode(&config); err != nil {
		return make(map[string]interface{})
	}
	return config
}

// GetHost returns the configured host or default.
func GetHost() string {
	config := LoadConfig()
	if host, ok := config["host"].(string); ok && host != "" {
		return host
	}
	return DefaultHost
}

// GetPort returns the configured port or default.
func GetPort() int {
	config := LoadConfig()
	if port, ok := config["port"].(float64); ok {
		return int(port)
	}
	if port, ok := config["port"].(int); ok {
		return port
	}
	return DefaultPort
}

// GetTimeout returns the configured timeout or default.
func GetTimeout() int {
	config := LoadConfig()
	if timeout, ok := config["timeout"].(float64); ok {
		return int(timeout)
	}
	if timeout, ok := config["timeout"].(int); ok {
		return timeout
	}
	return DefaultTimeout
}

// GetBufferSize returns the configured buffer size or default.
func GetBufferSize() int {
	config := LoadConfig()
	if bufSize, ok := config["buffer_size"].(float64); ok {
		return int(bufSize)
	}
	if bufSize, ok := config["buffer_size"].(int); ok {
		return bufSize
	}
	return DefaultBufSize
}
